packages/substrate_metadata/lib/utils/utils.dart

```diff
@@ -67,3 +67,34 @@ String sha256(dynamic data) {
 
   return scale.encodeHex(bytes);
 }
+
+/// Returns normalized Json to Human readable format
+extension ToJson<T> on T {
+  T toJson() {
+    if (this is List<T>) {
+      return (this as List<T>).map((T e) => e.toJson()).toList() as T;
+    }
+    if (this is List<Map<String, dynamic>> || this is Map<String, dynamic>) {
+      return jsonDecode(toHuman());
+    }
+    return this;
+  }
+
+  String toHuman() {
+    if (this is List<Map<String, dynamic>> || this is Map<String, dynamic>) {
+      return jsonEncode(
+        this,
+        // handle BigInt and Some
+        toEncodable: (value) {
+          if (value is BigInt) {
+            return value.toString();
+          } else if (value is scale.Some) {
+            return value.toString();
+          }
+          return value;
```

Comment:
Can't we have just `value.toString()`? Also, can't we do this in a single statement
```suggestion
          if (value is BigInt || value is scale.Some) {
            return value.toString();
          }
          return value;
```

Reply:
Could be :
```suggestion
          return (value is BigInt || value is scale.Some) ? value.toString() : value;
```

Reply:
@justkawal can you give me an example of a extrinsic with Some using this toJson? I want to check how it appears

Reply:
It will appear as: `"Some(Some(123))"` or `"Some(None)"`.

Here: https://github.com/rankanizer/polkadart/blob/930fec67fa737bb234e90ffac8f8c99d70bac2c1/packages/polkadart_scale_codec/lib/src/core/option_type.dart#L51

but I'm planning to change that and remove word `Some` from the string in the upcoming PR.

Reply:
Ok not good. In my opinion we should make it just like an enum so it should be something like this:
`{some_key: {None: null}}}`or `{some_key: {Some: 10}}` or `{some_key: {Some: null}}`

Reply:
Obviously you need to check recursively as we could have Some<Some<Some<T>>>

Reply:
So after removing word it would be evaluated like:

```dart
@override
String get toString() {
    if (value is _Option) {
        return (value as _Option).value.toString();
    }
    return value.toString();
}
```

So: `Some(Some(Some(123)))` would evaluated to `"123"`
So: `Some(Some(None))` would evaluated to `"None"`

Reply:
It is important to know if it was Some and not just the final value. For example when decoding extrinsic, I want to know the exact arg.

Reply:
Like: `Some(Some(Some(123)))`  ->  `{ "some_key": { "Some": { "Some": { "Some": 123 } } } }` ?

Reply:
Yep, I believe that's the way to go, what do you think @Lohann

Reply:
It might get confused with a dictionary/custom data structure. It's fine for encoding, because you know the type you're encoding to, but you couldn't differ nested optional values from a dictionary while decoding. Most solutions solve this reserving a keyword and documenting that this conflict might happen if the user tries to use reserved keys while decoding.

Reply:
Real solution for this would be to refactor everything to make it strongly typed, but it's probably too late for that

Reply:
That's for the "json" output. I believe we do have the types when using default dart objects, don't we?
But yeah in theory it could be a bit confusing if using the reserved keys, for example, decoding the String Some: `{some_key: "Some"}` though it is important to know the context here, that to decode anything you need to know the metadata. So you would already know that's an enum and not a string. Because otherwise you can't even decode it.

Reply:
Exactly, also I don’t see a future problem arising from it.

Reply:
@leonardocustodio  I would also give something like: `Some.fromJson(Map<String, dynamic> map)` which would help to create the Some or None Object back.